src/data/energy.ts

```diff
@@ -348,31 +347,22 @@ const getEnergyData = async (
   end?: Date,
   compare?: boolean
 ): Promise<EnergyData> => {
-  const [configEntries, info] = await Promise.all([
-    getConfigEntries(hass, { domain: "co2signal" }),
-    getEnergyInfo(hass),
-  ]);
-
-  const co2SignalConfigEntry = configEntries.length
-    ? configEntries[0]
-    : undefined;
+  const info = await getEnergyInfo(hass);
 
   let co2SignalEntity: string | undefined;
-  if (co2SignalConfigEntry) {
-    for (const entity of Object.values(hass.entities)) {
-      if (entity.platform !== "co2signal") {
-        continue;
-      }
-
-      // The integration offers 2 entities. We want the % one.
-      const co2State = hass.states[entity.entity_id];
-      if (!co2State || co2State.attributes.unit_of_measurement !== "%") {
-        continue;
-      }
+  for (const entity of Object.values(hass.entities)) {
+    if (entity.platform !== "co2signal") {
```

Comment:
Alright this is weird. We were doing this all wrong. I think the proper way we should do this is to use the config entry and only its entities. Otherwise this gets messy if someone has a "co2signal" that they don't want to use in the energy dashboard (edgy, I know) or more likely they can have 2+ "co2signal" integrations to monitor different regions and only 1 is actually configured for the energy dashboard.

Reply:
It already is, as we would just use the first config entry for co2signal we found?

Reply:
What if I have 2 config entries because I want to monitor the CO2 in Germany for example? We are assuming there will be only 1 config entry but we allow the user to add many.
I doubt this PR will break anything because it already works like this but I don't like that it does.

Reply:
Yes, I agree with you on that, and it could be that it picks another entity now even with the updated logic.

But that should be a separate PR, as it needs core support etc.